Tests/SonarScanner.MSBuild.Shim.Test/JsonPropertiesWriterTest.cs

```diff
@@ -42,4 +46,846 @@ public void Constructor_LoggerIsNull_ThrowsOnNullArgument()
+            AnalysisResults = [],
+            ProjectLanguage = language,
+            Encoding = encoding
+        };
+        if (coverageReportPath is not null)
+        {
+            projectInfo.AddAnalyzerResult(AnalysisType.VisualStudioCodeCoverage, coverageReportPath);
+        }
+        if (files is not null && files.Any())
+        {
+            fileListFilePath.Should().NotBeNullOrWhiteSpace("Test setup error: must supply the managedFileListFilePath as a list of files has been supplied");
+            File.WriteAllLines(fileListFilePath, files.Select(x => x.FullName));
+            projectInfo.AddAnalyzerResult(AnalysisType.FilesToAnalyze, fileListFilePath);
+        }
+        return projectInfo;
+    }
+
+    private static FileInfo CreateEmptyFile(string parentDir, string fileName) =>
+        new(CreateFile(parentDir, fileName, string.Empty));
+
+    private static string CreateFile(string parentDir, string fileName, string content)
+    {
+        var fullPath = Path.Combine(parentDir, fileName);
+        File.WriteAllText(fullPath, content);
+        return fullPath;
+    }
+
+    private static ProjectData CreateTestProjectDataWithPaths(string language, string[] analyzerOutPaths = null, string[] roslynOutPaths = null, string[] telemetryPaths = null)
+    {
+        analyzerOutPaths ??= [];
+        roslynOutPaths ??= [];
+        telemetryPaths ??= [];
+        var projectData = new ProjectData(new ProjectInfo
+        {
+            ProjectGuid = new Guid("5762C17D-1DDF-4C77-86AC-E2B4940926A9"),
+            ProjectLanguage = language
+        });
+        foreach (var path in analyzerOutPaths)
+        {
+            projectData.AnalyzerOutPaths.Add(new FileInfo(path));
+        }
+        foreach (var path in roslynOutPaths)
+        {
+            projectData.RoslynReportFilePaths.Add(new FileInfo(path));
+        }
+        foreach (var path in telemetryPaths)
+        {
+            projectData.TelemetryPaths.Add(new FileInfo(path));
+        }
+        return projectData;
+    }
+
+    // ToDo: https://sonarsource.atlassian.net/browse/SCAN4NET-747
+    // this method should be moved/re-implemetented to its own class "JsonPropertiesUtils as it seems to be needed for the PropertiesFileGeneratorTests
+    private static List<KeyValuePair<string, string>> JsonPropertiesReader(JArray jsonProperties)
+    {
+        var result = new List<KeyValuePair<string, string>>();
+        foreach (var item in jsonProperties)
+        {
+            if (item is JObject jsonObject)
+            {
+                // jsonObject.Properties() will return an array of key-value pairs
+                // [0]: { "key": "7B3B7244-5031-4D74-9BBD-3316E6B5E7D5.sonar.projectKey"}
+                // [1]: { "value": "7B3B7244-5031-4D74-9BBD-3316E6B5E7D5"}
+                var propertyKey = jsonObject.Properties().First().Value.ToString();
+                var propertyValue = jsonObject.Properties().Last().Value.ToString().Trim('\"');
```

Comment:
I need the trim for the values added [here](https://github.com/SonarSource/sonar-scanner-msbuild/blob/d2dc549eaad6b471d5d7e1226ec8aa31d4e7fec7/src/SonarScanner.MSBuild.Shim/JsonPropertiesWriter.cs#L104).
For some reason when I pull them here it puts them again in double quotes.
I tried to store them in different ways but I still have the issue.

Since this method will be re-written in the helper class I'd rather not push further in this.

Reply:
Removed